### PR TITLE
feat: auto-build (with public logs)

### LINKED ISSUE #42: show build logs in PR
See @jlhood's awesome serverless app: https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs

### PR DESCRIPTION
**Pull request by eladb**
If the `autoBuild` flag is set, a CodeBuild project
will be created and configured to automatically build
any commits to the repo, including GitHub pull requests.

If the `publicLogs` flag is also set, build logs will be published
to S3 and also a comment will automatically be posted on the GitHub
PR with a link.

Fixes #42

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by eladb** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://7x77720gqi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b1d6214c-89f0-4d1d-b7d6-ae44aa74e81b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by eladb** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://7x77720gqi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=058c9fa3-fbb4-419d-bcef-bcf04af8a7e8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by eladb** [comment]
Copy: @jlhood

**Comment by eladb** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://7x77720gqi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0bcd387e-355e-40cf-aae4-6dba3b44e2c5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: lib/auto-build.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    const project = new codebuild.Project(this, 'Project', {
+      source: props.repo.createBuildSource(this, true),
+      environment: createBuildEnvironment(props.environment),
+      badge: true,
+      buildSpec: props.buildSpec
+    });
+
+    // not support in this version of the cdk
+    const cfnProject = project.node.tryFindChild('Resource') as Resource;
```
Comment (rix0rrr) [inline lib/auto-build.ts]: Can we use `defaultChild` yet?

File: lib/auto-build.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      FilterGroups: [
+        [ { Type: 'EVENT', Pattern: 'PUSH,PULL_REQUEST_CREATED,PULL_REQUEST_UPDATED' } ]
+      ]
+    });
+
+    const publicLogs = props.publicLogs !== undefined ? props.publicLogs : false;
+    if (publicLogs) {
+      new serverless.CfnApplication(this, 'GitHubCodeBuildLogsSAR', {
+        location: {
+          applicationId: 'arn:aws:serverlessrepo:us-east-1:277187709615:applications/github-codebuild-logs',
```
Comment (rix0rrr) [inline lib/auto-build.ts]: Does this require that we run the build itself in `us-east-1` as well?

File: lib/pipeline.ts [hunk 3]
```diff
@@ -165,6 +184,10 @@ export class Pipeline extends cdk.Construct {
 
     // emit an SNS notification every time build fails.
     this.addBuildFailureNotification(buildProject, `${props.title} build failed`);
+
+    if (props.autoBuild) {
```
Comment (rix0rrr) [inline lib/pipeline.ts]: Not entirely clear to me what an "autobuild" is and why it needs a second build project.

File: lib/pipeline.ts [hunk 4]
```diff
@@ -293,6 +316,18 @@ export class Pipeline extends cdk.Construct {
     });
   }
 
+  /**
+   * Enables automatic builds of
```
Comment (rix0rrr) [inline lib/pipeline.ts]: of...?

File: lib/auto-build.ts (hunk 1 above)
Comment (eladb) [reply lib/auto-build.ts]: Doesn't exist in this version of the CDK

File: lib/auto-build.ts (hunk 2 above)
Comment (eladb) [reply lib/auto-build.ts]: Nope. SAR takes care of replication.

File: lib/pipeline.ts (hunk 3 above)
Comment (eladb) [reply lib/pipeline.ts]: The first build project source is the pipeline and this one is github. It's a different build project...

Comment (rix0rrr) [reply lib/pipeline.ts]: Aha, so maybe call it something like `PRBuild` ?

Comment (eladb) [reply lib/pipeline.ts]: Technically it’s not just for PRs. It also builds any commit to master and in the future we can add some configuration to control when it is triggered. It is automatic builds for all intents and purposes.

File: lib/auto-build.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+          semanticVersion: '1.0.3'
```
Comment (jlhood) [inline lib/auto-build.ts]: Sorry, the README is out of date. Latest version is 1.0.4 which adds throttling to the API GW created by the app. This is good to protect you from abusers who could call the API at a high rate resulting in a high AWS bill so I recommend doing the update. I'll update the app README to reference the latest version.

**Pull request merged**